rcl_lifecycle/test/test_rcl_lifecycle.cpp

```diff
@@ -133,14 +133,19 @@ TEST(TestRclLifecycle, lifecycle_transition) {
 
   ret = rcl_lifecycle_transition_init(
     &transition, expected_id, &expected_label[0], nullptr, nullptr, &allocator);
-  EXPECT_EQ(ret, RCL_RET_OK);
-  rcutils_reset_error();
+  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
```

Comment:
I think we still have to reset the error message after this. Basic rule: Every time we consume the error message, we reset it.

Reply:
same below